### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by AntonioGomez12**
website link: https://orange-flower-0da9ed11e.1.azurestaticapps.net

### CODE REVIEW

**Comment by NazariyPogo** [comment]
I will review

File: .github/workflows/azure-static-web-apps-orange-flower-0da9ed11e.yml [hunk 1]
```diff
@@ -1,61 +1,56 @@
-name: Azure Static Web Apps CI/CD
-
```
Comment (NazariyPogo) [inline .github/workflows/azure-static-web-apps-orange-flower-0da9ed11e.yml:2]: Adding env:
  API_URL: ${{secrets.yourSecretName}}
here like done in class will allow you to simplify the url entered in the axios call

**Review by NazariyPogo — COMMENTED** [review]
1. Create a leaderboard API Controller ❌✅

Create an endpoint that returns the top 10 scores (HttpGet) ✅
Results should include Name, average guesses, and number of games played ✅
Create an endpoint that allows for posting a score (HttpPost) with data in the body ✅
A new name should add a new record ✅
An existing name should update that record ✅
Save and load all data with Entity Framework to an Azure SQL database or LocalDb for debugging ✅
Class should be named "Player" and "Players" for the DbSet ✅ //named playersStats, but should be fine
Fields should be:
int PlayerId ✅
string Name ✅
int GameCount ✅
double AverageAttempts ✅
int AverageSecondsPerGame ✅(OPTIONAL)
The logic should reside in a service that is injected via dependency injection into the controller ✅
Set up CORS (Cross Origin Resource Sharing) ❌
2. Create a dialog for getting the user's name ✅ //I like your take on account creation

If we don't have the user's name, present the user with a dialog that allows them to type in their name ❌ //Option available but not presented on end of game
The users current name should show on the game page (upper right) ✅
This should be visible when setting/changing the name. It should change as the user types ✅
The user should be able to click on their name to change it (with the same dialog) ✅
Save the name in local storage and load it for the next game if it is set ❌ //Name resets every time I refresh
If the name is not set, display Guest as the name. ✅
Clicking on this allows for preemptive setting of the name before the end of the game ✅
Dialog must be persistent, doesn't close by clicking off the dialog ✅
3. Save the Score and Seconds to complete ✅

Save the score and time to complete in seconds at the end of the game if we have a name. ❌ //I dont see any posts after a game is over in the network
If the user hasn't set a name, save the score and seconds as "Guest". (Be sure to prompt first) ❌ //No prompt
Save should be done with an Axios Post ❌ //Needs to be finished
4. Create a Leaderboard Page ✅ //Looks good as a draft, just need records to confirm

Create a /leaderboard page in the application that shows the top scores ❌✅ //No records yet
Use Axios to call the leaderboard API ❌ // calls need to be fixed
Make a nice display of the returned data ✅ //looks like it'll be pretty
Add a link on the front page to access the leaderboard ✅
Add a link to the app bar to access the leaderboard ❌ //Need link in app bar to leaderboard
5. Unit testing ✅ //Tests look good

Tests for the service ❌✅
Integration tests for the controller ❌ (OPTIONAL)
6. Deploy site to Azure ✅

Deploy API site with Github Action to Azure Web App ✅
Unit Tests should run on build ✅
Add API Base URL replacement to front end project via secret ❌ //Pls do this like we did in class, helped me a lot
Extra Credit
Unit testing on the client side (3) ❌
Animate the leaderboard (3) ❌
Create a logo (3) ❌✅
Add Average Seconds to win in score (3) ✅

**Comment by NazariyPogo** [comment]
One more thing... That click noise... interesting choice... seems like spamming it should come with a warning lol

**Comment by ZackWard-student** [comment]
would be nice to have a link to your swagger server api as well.

**Comment by ZackWard-student** [comment]
wordle-web/components/hsquid.png should probably be in the the static folder. Also it is a cursed image and should be removed from the project. (It also made my wife do a double take and question what I was doing)

**Comment by ZackWard-student** [comment]
Neither front end nor back end compile...

**Review by ZackWard-student — CHANGES_REQUESTED** [review]
I like a lot of the ideas, but as is it does not compile.

**Comment by ZackWard-student** [comment]
I am not sure what dbs you had stood up but there where no migration files for me to stand up the same ones.

**Comment by ZackWard-student** [comment]
Don't forget to at least make it compile and run your linter before Meg sees it!

File: .github/workflows/azure-static-web-apps-orange-flower-0da9ed11e.yml [hunk 2]
```diff
@@ -1,61 +1,56 @@
...
-    runs-on: ubuntu-latest
-    name: Close Pull Request Job
-    steps:
-      - name: Close Pull Request
-        id: closepullrequest
-        uses: Azure/static-web-apps-deploy@v1
-        with:
-          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN }}
-          action: "close"
+name: Azure Static Web Apps CI/CD
```
Comment (mmwoodfo) [inline .github/workflows/azure-static-web-apps-orange-flower-0da9ed11e.yml:1]: It doesn't seem like your API is deployed correctly. Getting 404s on the website.

File: Wordle.api/Data/AppDbContext.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+{
+    public class AppDbContext : DbContext
+    {
+        public AppDbContext(DbContextOptions<AppDbContext> options)
+            : base(options)
+        {
+        }
+
+        public DbSet<ScoreStat> ScoreStats { get; set; } = null!;
+        public DbSet<Player> PlayerStats { get; set; } = null!;
```
Comment (mmwoodfo) [inline Wordle.api/Data/AppDbContext.cs:15]: Looks like you don't have a player class?

**Review by mmwoodfo — COMMENTED** [review]
Your backend code doesn't run because you're missing a Player class and the nuget package for entity framework. Additionally, the PR is technically a few minutes late, but I won't count that against you.

**Pull request closed without merging**